### PR TITLE
Ridge elastic net

### PR DESCRIPTION
**Pull request by LElgueddari**
This PR implements the Elastic Net and the Ridge proximity operator used as regularization when the data is highly correlated

### CODE REVIEW

File: modopt/opt/proximity.py [hunk 1]
```diff
@@ -546,3 +547,138 @@ def _cost_method(self, *args, **kwargs):
...
+            Additional multiplication factor
+
+        Returns
+        -------
+        np.ndarray thresholded data
+
+        """
+
+        soft_threshold = self.beta * extra_factor
+        print(soft_threshold)
```
Comment (sfarrens) [inline modopt/opt/proximity.py]: Please remove print statement.

File: modopt/opt/proximity.py [hunk 2]
```diff
@@ -546,3 +547,138 @@ def _cost_method(self, *args, **kwargs):
...
 
         return cost_val
+
+
+class Ridge(ProximityParent):
+    """L2-norm proximity operator (i.e. shrinkage)
+
+    This class defines the L2-norm proximity operator
+    prox(y) = argmin 0.5||x-y||_2^2 + alpha*||x||_2^2
+              x in C
```
Comment (sfarrens) [inline modopt/opt/proximity.py]: Not necessary, but if you wanted you could have this equation render in LaTeX for the API docs. See *e.g.* [this example](https://cea-cosmic.github.io/ModOpt/modopt.math.matrix.html#modopt.math.matrix.nuclear_norm).

File: modopt/opt/proximity.py [hunk 3]
```diff
@@ -546,3 +547,138 @@ def _cost_method(self, *args, **kwargs):
...
+        return cost_val
+
+
+class ElasticNet(ProximityParent):
+    """Linear combination between L2 and L1 norm proximity operator,
+    described in [Z2005]
+
+    This class defines the L2-norm proximity operator
+    prox(y) = argmin 0.5||x-y||_2^2 + alpha*||x||_2^2 + beta*||x||_1
+              x in C
```
Comment (sfarrens) [inline modopt/opt/proximity.py]: See above comment.

File: modopt/opt/proximity.py (hunk 1 above)
Comment (LElgueddari) [reply modopt/opt/proximity.py]: Done

File: modopt/opt/proximity.py (hunk 2 above)
Comment (LElgueddari) [reply modopt/opt/proximity.py]: Replaced

File: modopt/opt/proximity.py (hunk 3 above)
Comment (LElgueddari) [reply modopt/opt/proximity.py]: Done

**Pull request merged**